include/cxx.h

```diff
@@ -30,6 +30,54 @@ template <typename T>
 class impl;
 }
 
+#ifndef CXXBRIDGE1_TRAITS
+#define CXXBRIDGE1_TRAITS
+namespace detail {
+template <typename... Ts>
+struct make_void {
+  using type = void;
+};
+
+template <typename... Ts>
+using void_t = typename make_void<Ts...>::type;
+
+template <typename Void, template <typename...> class, typename...>
+struct detect : std::false_type {};
+template <template <typename...> class T, typename... A>
+struct detect<void_t<T<A...>>, T, A...> : std::true_type {};
+
+template <template <typename...> class T, typename... A>
+using is_detected = detect<void, T, A...>;
+
+template <typename T>
+using detect_IsRelocatable = typename T::IsRelocatable;
+template <typename T>
+struct get_IsRelocatable
+    : std::is_same<typename T::IsRelocatable, std::true_type> {};
+
+template <typename T>
+using detect_IsCloneable = typename T::IsCloneable;
+template <typename T>
+struct get_IsCloneable : std::is_same<typename T::IsCloneable, std::true_type> {
+};
+} // namespace detail
+
+template <typename T>
+struct IsRelocatable
```

Comment:
I wanted to share some of the detection bits here with `IsRelocatable` to make `Box::clone` conditionally compiled. I collapsed these into a single "traits" section, but if you'd prefer, I can keep "IsCloneable" separate